### PR TITLE
Fix serialization of CalcLengthOrPercentage

### LINKED ISSUE #15168: `calc()` value should not be simplifed to be bare value for specified values
For example, declaration `width: calc(2px)` should be serialized to `width: calc(2px)` rather than `width: 2px`, because `width: calc(-2px)` is valid while `width: -2px` is not.

Serialization of `calc()` is undefined in the spec at the moment, but the bottom line is that, this matches behavior of all major browsers (Gecko, Blink, Edge, and presumably also WebKit).

### PR DESCRIPTION
**Pull request by canova**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15168 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15178)

### CODE REVIEW

File: components/style/values/specified/length.rs [hunk 1]
```diff
@@ -798,12 +784,7 @@ impl ToCss for CalcLengthOrPercentage {
             };
         }
 
-        let count = count!(ch, em, ex, absolute, rem, vh, vmax, vmin, vw, percentage);
-        assert!(count > 0);
```
Comment (canaltinova) [inline components/style/values/specified/length.rs]: This assert was unnecessary because `CalcLengthOrPercentage` fails when it's empty.

**Comment by emilio** [comment]
Tests?

**Comment by emilio** [comment]
Oh, @jdm added the label before I commented :)

**Comment by Wafflespeanut** [comment]
Do the tests in `tests/wpt/mozilla/calc.html` pass?

**Comment by canaltinova** [comment]
@Wafflespeanut Interesting, I'm getting openssl error when I try to test this. I'll try clean build and try again.

**Comment by canaltinova** [comment]
Fixed tests in `tests/wpt/mozilla/calc.html`. But I think we need to wait for #15164 until we can run try and look for other failing/passing tests.

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /_mozilla/mozilla/calc.html:
  │ FAIL [expected PASS] calc for border-spacing
  │   → assert_equals: expected "1px 1px" but got "calc(1px) calc(1px)"
  │ 
  │ @http://web-platform.test:8000/_mozilla/mozilla/calc.html:147:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:501:9
  │ @http://web-platform.test:8000/_mozilla/mozilla/calc.html:145:5
  └ @http://web-platform.test:8000/_mozilla/mozilla/calc.html:144:1
```

**Comment by canaltinova** [comment]
Updated test and added a test for empty `calc()`.

**Comment by jdm** [comment]
@bors-servo: retry
* #15227

**Pull request merged**